src/Array.ts

```diff
@@ -735,29 +734,34 @@ export const rotate = <A>(n: number, xs: Array<A>): Array<A> => {
   }
 }
 
+/**
+ * Test if a value is a member of an array
+ * @function
+ * @since 1.3.0
+ */
+export const member = <A>(S: Setoid<A>) => (as: Array<A>, a: A): boolean => {
+  return findIndex(as, e => S.equals(e, a)).isSome()
```

Comment:
This operation is very useful and can be used a lot, can we get ride of the temporary Option to reduce the extra allocation / GC pressure?

```typescript
export const member = <A>(S: Setoid<A>) => (as: Array<A>, a: A): boolean => {
  const len = as.length
  const predicate = (e: A) => S.equals(e, a)
  for (let i = 0; i < len; i++) {
    if (predicate(as[i])) {
      return true
    }
  }
  return false
}
```

Reply:
@sledorze agreed